### ISSUE TITLE
Changes to storefront SEO routing/links or Liquid breaks Shopify theme

### ISSUE THREAD

**Issue by Esbob** [VirtoCommerce/vc-platform#440]
Hi,

After installing version 2.10.1425, my Shopify theme is totally messed up: Missing colors, wrong fonts, missing collection sidebars. It worked just fine in 2.9.
Have you made changes to the SEO routing and links that are not backward compatible? Are you still compatible with Shopify Liquid themes? I'm particularly thinking about the way categories are interpreted as collections.
I set my store to use short SEO links, hoping that would correspond to how it was working before version 2.10, but it didn't fix the problem.
I'm using the Azure blob storage provider to store themes and somehow your own Matherial theme seems to display correctly, so I don't think the error is due to inaccessible assets or templates.

Do you have any suggestions for what I could try in order to identify the source of the error? 

By the way, how do I disable caching of Liquid templates? There used to be a button for that...

Regards,
Esbob

**Comment by tatarincev** [comment]
For begin can you check network traffic for your theme to  detect which request failed (css, js)?  
We do not change shopify themes compatibility all should works as was.

**Comment by Esbob** [comment]
Using Fiddler, I can see that all requests for resources are OK (200)...
To make it a little easier to debug my Liquid templates, how do I disable caching?

**Comment by Esbob** [comment]
It seems to be a caching problem: When I request an image e.g. http://mysite/themes/assets/slideshow_2.jpg, I'm actually returned another picture with a similar name, namely http://mysite/themes/assets/lb_slideshow_2.jpg. 
If I delete this last image from blob storage and restart the web site (i.e. clearing the cache), I'm returned the correct image. I think I'm incorrectly getting a cache hit - can this be disabled?

**Comment by tatarincev** [comment]
To disable full page caching need to make this changes in storefront web.config

```
   <system.web>
       <caching> 
          <outputCache defaultProvider="CacheManagerOutputCacheProvider" enableOutputCache="**false**">
```

To disable themes templates  and API request caching need to change  cache expiration time to small values 
```

 <cacheManager >
        <managers>
 <cache name="storefrontCache" updateMode="Up" enableStatistics="false" enablePerformanceCounters="false">
                <handle ref="runtimeMemory" name="inMemoryCacheHandle" expirationMode="Absolute" timeout="**1s**" />
            </cache>
```

**Comment by Esbob** [comment]
Thanks - I'll give it a try :-)

**Comment by Esbob** [comment]
OK, so my problem was related to Azure blob storage - things work fine with local storage. 
So, either I'm not configuring it correctly or there is a bug in the Azure blob storage theme item retrieval. 
I tried to follow the guidelines described in the VC 2.10 release notes for how to use Azure blob theme storage, 
but instead of adding connection strings to the two web.config files, I created two connection string entries in the 
Azure Portal -> App Services -> [MyAppService] -> Application Settings -> Connection Strings:
**CmsContentConnectionString** with value "provider=AzureBlobStorage;DefaultEndpointsProtocol=https;AccountName=yourAccountName;AccountKey=yourKey;rootPath=cms-content"
**ContentConnectionString** with same value as above (the connection strings in the release notes seem to be equivalent)

This apparently worked fine with the sample Clothing shop, but my shop had problems with liquid templates not being parsed and executed, and files not being returned.
For now, I've unfortunately had to drop the Azure blob theme storage.
